[39497] Makes action dispatch methods mutable

willypuzzle opened the pull request:
Refactors action dispatch methods to take a mutable reference to the handler, allowing modification of the session state during action execution.

This change avoids unnecessary borrowing and cloning of data, and it aligns better with the intended behavior of modifying the session's input state during action dispatching.


Testing: This task is only for refactoring, it doesn't require new tests
Fixes: #39497

willypuzzle committed:
Makes action dispatch methods mutable

Refactors action dispatch methods to take a mutable reference to the
handler, allowing modification of the session state during action
execution.

This change avoids unnecessary borrowing and cloning of data, and it
aligns better with the intended behavior of modifying the session's
input state during action dispatching.

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

mrobinson commented on components/webdriver_server/actions.rs:
```diff
@@ -1,7 +1,6 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
-
```
Looks like you've accidentally removed this? Please restore it.

mrobinson commented:
Don't forget to run `./mach fmt` and also to sign your commit, but otherwise this is looking pretty good. I can give it a more thorough look once it's passing Lint and DCO checks. Thanks!

willypuzzle committed:
Improves code formatting and readability

Applies minor formatting adjustments to improve code
readability and consistency, specifically around function
definitions and chained method calls. No functional changes
are included.

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

willypuzzle committed:
Simplifies action dispatch logic.

Refactors the action dispatch logic for pointer events
to improve readability by removing unnecessary scoping.

This change does not affect the functionality, but it
makes the code easier to understand and maintain.

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

willypuzzle commented on components/webdriver_server/actions.rs:
```diff
@@ -354,22 +352,28 @@ impl Handler {
+        let x: f32;
+        let y: f32;
+        {
+            let pointer_input_state = self.get_pointer_input_state(source_id);
```
you're right, the brackets were useless, I removed them.

yezhizhen commented on components/webdriver_server/actions.rs:
```diff
@@ -370,6 +366,10 @@ impl Handler {
         // Step 6. Add button to the set corresponding to source's pressed property
         pointer_input_state.pressed.insert(action.button);
+
+        let x = pointer_input_state.x as f32;
+        let y = pointer_input_state.y as f32;
```
Why not remove these and keep rest unchanged.

willypuzzle replied:
I tried to remove the variables assignment and keep the original use but the compiler complains

yezhizhen commented on components/webdriver_server/actions.rs:
```diff
@@ -379,8 +379,8 @@ impl Handler {
             action.button.into(),
-            pointer_input_state.x as f32,
-            pointer_input_state.y as f32,
+            x,
+            y,
```
We can keep these unchanged.

willypuzzle replied:
As I explained above if I maintain the original reference the compiler complains

yezhizhen commented on components/webdriver_server/actions.rs:
```diff
@@ -581,8 +574,11 @@ impl Handler {
-                pointer_input_state.x = x;
-                pointer_input_state.y = y;
+                {
+                    let pointer_input_state = self.get_pointer_input_state(source_id);
```
Why we `get_pointer_input_state` again? We already did this at the beginning.

willypuzzle replied:
If I use the above reference I get an error from the compiler

yezhizhen commented on components/webdriver_server/session.rs:
```diff
@@ -121,16 +120,16 @@ impl WebDriverSession {
 
-    pub fn input_state_table(&self) -> Ref<'_, HashMap<String, InputSourceState>> {
-        self.input_state_table.borrow()
+    pub fn input_state_table(&self) -> &HashMap<String, InputSourceState> {
+        &self.input_state_table
     }
```
I wonder if we should just make the corresponding field instead of using these trivial Getter.

cc @mrobinson

mrobinson replied:
Are they only used internally? If so, I would say just use the field, if used externally, I would make these `pub(crate)`.

yezhizhen replied:
I mean make the field `pub(crate)`.

yezhizhen reviewed (changes requested):
> Testing: This task is only for refactoring, it doesn't require new tests

This PR **does change the behaviour**. You are keep reassigning static `p_x` and `p_y` to `current_x`, which would remove the smooth behaviour of "pointermove". And I see a few workarounds to get around borrow checkers which seem questionable.

There are a few other new features PR for webdriver ongoing, so this should probably wait. I'll put up another new PR later to finish this task, and add you as co-author. Thanks!
